src/graphforge/datasets/loaders/ldbc.py

```
@@ -0,0 +1,240 @@
+            raise FileNotFoundError(f"Dataset path not found: {path}")
+
+        # Extract archive if needed
+        if is_compressed_archive(path):
+            # Extract to temporary directory
+            extract_dir = path.parent / f".{path.stem}_extracted"
+            extract_archive(path, extract_dir)
+            data_dir = extract_dir
+        elif path.is_dir():
+            data_dir = path
+        else:
+            raise ValueError(f"Expected .tar.zst archive or directory, got: {path.suffix}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)